### ISSUE TITLE
Missing editor in proceedings causes duplicated title

### ISSUE THREAD

**Issue by jnv** [michal-h21/biblatex-iso690#58]
I work with conference proceedings from ACM which usually don't have an editor. This seems to cause an interesting problem with `iso-authoryear` format. Given the following database entry:

```
@INPROCEEDINGS{prispevekBezEditora,
title = {Velmi zajímavý příspěvek},
booktitle = {Konference bez editora},
year = {2007},
langid = {czech},
location = {Praha},
publisher = {Academia},
isbn = {978-222-626-222-2},
pages = {22-45},
author = {Vlaštovka, Josef},
}
```

results in the following output:

![selection_2017-06-17_02](https://user-images.githubusercontent.com/616767/27252023-280eedaa-5354-11e7-8b97-2c8f9828782d.png)

Notice the duplication of _Velmi zajímavý příspěvek_ in what's supposed to be the booktitle. Unlike in `collection` editor is optional for `proceedings` so this should be a correct entry. But adding an editor seems to fix the problem.

**Comment by jnv** [comment]
I have prepared an example case here: https://github.com/jnv/biblatex-iso690/commit/cb50e9c86ca5f0f5069a364b151c99448b2d66cc

**Comment by jnv** [comment]
Also, this seems to happen only in `authoryear` format. When I switch to `numeric`, it looks fine:

![selection_2017-06-17_03](https://user-images.githubusercontent.com/616767/27252241-79a8f110-5359-11e7-86ff-d232068fd6cf.png)

**Comment by jnv** [comment]
Okay, so this looks like to be caused by the alternate branch in `bbx:editor` macro, which invokes `labeltitle` macro. I wonder if it is needed to redeclare `editor` macro at all; it is not present in other styles so I guess it was put there for the dashcheck.

**Comment by michal-h21** [comment]
Try to put `\renewbibmacro{labeltitle}{}` to your document preamble. I think that this macro is unnecessary, but we should wait for @DavidLuptak if he remembers what was it's purpose.

**Comment by jnv** [comment]
Thanks, in the end I made a custom bbx style, because I needed to cite a movie (i.e. work without a primary author). Removing the redeclaration of `editor` macro did not cause any problem for me, AFAIK.

**Comment by DavidLuptak** [comment]
@jnv Good catch!

> I wonder if it could be related to #57...

I don't think the commit solving this issue caused this behavior.

Well, yes, you are right, the problem is in the `editor` macro in the `iso-authoryear` style. The `labeltitle` macro is used there because of the following the `biblatex` documentation.. the field `label` is used as a fallback for any missing data (see section 2.3.2 for more).
However, it is true that it is not an appropriate behavior for our style, maybe because the field `label` is hardly ever used in the bib databases?
But the redefinition of the macro `editor` in `iso-authoryear` is needed since it resolves the stuff with dates/years which is there by default.

**Comment by moewew** [comment]
I think the problem here is that the `editor` macro is used after `in:` for `@incollection`. The `authoryear` and `authortitle` bibliography styles both redefine `editor` to allow for a dash. A dash is, however, not wanted after the 'in:'. Additionally, `authoryear` also redefines `editor` in such a way to display the title of the work if no editor is present in order to show something before the year. This is probably wanted at the very beginning of an entry, but it is not suitable after the 'in:'.

With all that in mind, I'd recommend to define a new macro that prints the editor after the 'in:' for `@incollection` and not rely on the standard `editor` bibmacro. That macro could be written in such a way that it does not check for dashes (and does not influence the check either) and would not print a replacement string in case the editor is missing. It could also be written in such a way that it does not obey the `\ifuseeditor` toggle, which may or may not be useful.

This would probably allow to retain the standard definitions of `editor` and let the dash be controlled by `biblatex`s options. If the dash is undesirable for ISO style, simply `\ExecuteBiblkiographyOptions{dashed=false}` in `iso-authoryear.bbx` and `iso-authortitle.bbx`.

**Comment by DavidLuptak** [comment]
The problem was in the `editor` macro as we said before, because `labeltitle` macro was executed from [within this macro](https://github.com/michal-h21/biblatex-iso690/commit/0e4eded3d7182d989b3d8cae7d142bbd9841e4aa#diff-414fa26031e84d4cee4820d5c948e6feL62).

The issue was completely fixed in c815202. No redefinition of `editor` macro is needed if we have correct macro for handling printing of author/editor/label/title before date.
